File: go-fiber/app/incoming/taskapi/task_request.go
package taskapi

import (
	"appfiber/outgoing/taskrepo"

	"github.com/google/uuid"
)

func (taskRequest TaskRequest) ToNewTaskEntity() taskrepo.TaskEntity {
	return taskrepo.TaskEntity{
		TaskId:      uuid.New(),
		Title:       taskRequest.Title,
		Description: taskRequest.Description,
	}
}

func (taskRequest TaskRequest) ToUpdatedTaskEntity(taskId uuid.UUID) taskrepo.TaskEntity {
	return taskrepo.TaskEntity{
		TaskId:      taskId,
		Title:       taskRequest.Title,
		Description: taskRequest.Description,
	}
}

File: java-javalin/app/src/main/java/science/monke/incoming/TaskRequest.java
package science.monke.incoming;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;
import science.monke.outgoing.Task;

public record TaskRequest(
    @JsonProperty("title") String title, @JsonProperty("description") String description) {
  public static Task toTask(final TaskRequest taskRequest) {
    return toTask(UUID.randomUUID(), taskRequest);
  }

  public static Task toTask(final UUID taskId, final TaskRequest taskRequest) {
    final Task task = new Task();
    task.taskId = taskId;
    task.title = taskRequest.title;
    task.description = taskRequest.description;
    return task;
  }
}
